src/components/Votes.jsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router";
+
+import { getVotesCount, updateArticleVotes } from "../api";
+
+const Votes = () => {
+  const { article_id } = useParams();
+
+  const [votesCount, setVotesCount] = useState(0);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    getVotesCount(article_id).then((votesCount) => {
+      setVotesCount(votesCount);
+    });
+  }, []);
+
+  const handleVote = () => {
+    setVotesCount((currentVotesCount) => currentVotesCount + 1);
+    setError(null);
+    updateArticleVotes(article_id).catch((err) => {
+      setVotesCount((currentVotesCount) => currentVotesCount - 1);
+      setError(
+        <span className="article-text-smol">
+          ❤︎ failed to register! Please try again!
+        </span>
+      );
+    });
+  };
+
+  return (
+    <span className="article-text-smol">
+      <button onClick={handleVote}>
+        ❤︎ {error ? { error } : null}
```

Comment:
the extra curly brackets around `error` on line 34 are yielding a runtime error when encountering a failed request because it is then treated as a JS object trying to render into a button, if you change the line to read:
```js
{error ? error : null }
```
then it should work as intended and apply the span correctly when the request fails 👍